ios/brave-ios/Sources/Brave/Frontend/Browser/Handlers/ReaderModeHandler.swift

```diff
@@ -24,12 +24,80 @@ public class ReaderModeHandler: InternalSchemeResponse {
       return nil
     }
     
+    // Decode the original page's response headers
+    var headers = [String: String]()
+    if let base64EncodedHeaders = _url.getQuery()["headers"]?.unescape(),
+       let data = Data(base64Encoded: base64EncodedHeaders),
+       let decodedHeaders = try? JSONSerialization.jsonObject(with: data) as? [String: String] {
+      headers = decodedHeaders
+    }
+    
+    headers = headers.filter({
+      let key = $0.key.lowercased()
+      
+      // These are the only headers kept from the original page
+      return key == "access-control-allow-origin" ||
+             key == "content-security-policy" ||
+             key == "strict-transport-security" ||
+             key == "content-language"
+    })
+    
+    // Tighten security by adding some of our own headers
+    headers["X-Frame-Options"] = "DENY"
+    headers["X-Content-Type-Options"] = "nosniff"
+    headers["Referrer-Policy"] = "no-referrer"
+    headers["Cache-Control"] = "private, s-maxage=0, max-age=0, must-revalidate"
+    
+    // Add Generic headers
+    headers["Content-Type"] = "text/html; charset=UTF-8"
+    
+    // Handle CSP header
+    // Must generate a unique nonce, every single time as per Content-Policy spec.
+    let setTitleNonce = UUID().uuidString.replacingOccurrences(of: "-", with: "")
+    
+    // Create our own CSPs
+    var policies = [
+      ("default-src", "'none'"),
+      ("base-uri", "'none'"),
+      ("form-action", "'none'"),
+      ("frame-ancestors", "'none'"),
+      //("sandbox", ""),                  // Do not enable `sandbox` as it causes `Wikipedia` to not work and possibly other pages
+      ("upgrade-insecure-requests", ""),
+      ("img-src", "*"),
```

Comment:
If the original page has `img-src` policy in a meta tag we should add it here instead of `*`

Reply:
Done. But I think if you append a second `img-src` that is not `*` it should take precedence over the `*`, iirc.